main.tf

```diff
@@ -1205,9 +1369,24 @@ resource "aws_route_table_association" "public" {
   count = var.create_vpc && length(var.public_subnets) > 0 ? length(var.public_subnets) : 0
 
   subnet_id      = element(aws_subnet.public.*.id, count.index)
-  route_table_id = aws_route_table.public[0].id
+  route_table_id = aws_route_table.public[var.enable_firewall ? count.index : 0].id
+}
+
+resource "aws_route_table_association" "public_internet_gateway" {
+  count = var.create_vpc && var.create_igw && length(var.firewall_subnets) > 0 ? 1 : 0
```

Comment:
IMO, Some loon (me) should be able to create the firewall subnets and leave `enable_firewall` set to false.  This needs to be changed to make that possible as `aws_route_table.internet_gateway` is not created unless `enable_firewall` is true.
```
  count = var.create_vpc && var.create_igw && var.enable_firewall && length(var.firewall_subnets) > 0 ? 1 : 0
```

Reply:
Thanks, i've made the changes to include it into the condition